blog/2024-05-16-blog-status-of-lbse-in-webkit.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Status of LBSE in WebKit"
+author: rbuis
+permalink: /blog/status-of-lbse-in-webkit.html
+thumbnail: /assets/lbse-logo-wide.png
+preview: A look back at the last 7 months of development on LBSE in WebKit.
+---
+
+This blog entry gives an update on upstreaming and development of LBSE in WebKit in the last 7 months. For an explanation of
+what LBSE is and how it is related to WPE see this previous <a href="https://wpewebkit.org/blog/05-new-svg-engine.html">entry</a> as a refresher.
+
+[Describe wix and its sponsoring]
+
+<figure>
+  <a href="/assets/svg/wix-logo.svg" target="_blank"><img
+    style="dsplay: block; margin-left: auto; margin-right: auto; height: 60px;"
+    src="/assets/svg/wix-logo.svg"
+    alt="Diagram showing a box for the WPE backend in between the UI process and WPEWebProcess">
+  </a>
+</figure>
+
+## Project kick off and WebKit Contributors Meeting
+
+In **October 2023** we started the project, mostly by thinking of the design and roadmap. Also we did some general SVG bug fixing. First, visual overflow computation for SVG renderers was <a href="https://commits.webkit.org/268981@main">corrected</a>, which fixed quite some SVG pixel tests. Various visual bugs were also fixed like <a href="https://commits.webkit.org/269034@main">unnecessary repainting when viewBox is used on &lt;svg&gt; elements</a> and <a href="https://commits.webkit.org/269360@main">incorrect clipping for outermost &lt;svg&gt; elements</a>
+
+Also in the same month, we attended the <a href="https://docs.webkit.org/Other/Contributor%20Meetings/ContributorMeeting2023.html">WebKit Contributors</a> meeting, where we presented a talk on the LBSE (slides are <a href="https://teams.pages.igalia.com/webkit/contributors-meeting-presentations/2023/igalia-slides/integrating-the-new-LBSE.html">here</a>) and the feedback on LBSE at the meeting was very positive. Giving the talk early on actually helped us since we needed to have a good design in place.
+
+## Supporting SVG resources
+
+The main focus in **October 2023** was introducing the SVG resource concept, as already outlined in the WebKit Contributors Meeting <a href="https://teams.pages.igalia.com/webkit/contributors-meeting-presentations/2023/igalia-slides/integrating-the-new-LBSE.html">talk</a>. Thus we started with <a href="https://commits.webkit.org/269522@main">adding the base SVG resource class</a>. The base class `RenderSVGResourceContainer` was kept as simple as possible with no support for resource invalidation or repainting logic. The main task of `RenderSVGResourceContainer` was to take care of registration so that the resource can be looked up by its clients.
+
+As the first SVG resource to implement we chose the SVG &lt;clip-path&gt;, so we <a href="https://commits.webkit.org/269635@main">landed RenderSVGResourceClipper</a>.
```

Comment:
Should `[RenderSVGResourceClipper](https://commits.webkit.org/269635@main)` maybe to put it in code blocks like the rest?

Reply:
Fixed, that is more consistent indeed.